parachain/pallets/ethereum-beacon-light-client/src/lib.rs

```diff
@@ -0,0 +1,666 @@
+//! # Ethereum 2 Light Client Verifier
+//!
+//! This module implements the `Verifier` interface. Other modules should reference
+//! this module using the `Verifier` type and perform verification using `Verifier::verify`.
+#![allow(unused_variables)]
+#![cfg_attr(not(feature = "std"), no_std)]
+
+#[cfg(test)]
+mod mock;
+
+#[cfg(test)]
+mod tests;
+mod merklization;
+
+use codec::{Decode, Encode};
+use frame_support::{dispatch::DispatchResult, log, transactional, Deserialize, Serialize};
+use frame_system::ensure_signed;
+use scale_info::TypeInfo;
+use sp_core::H256;
+use sp_runtime::RuntimeDebug;
+use sp_std::prelude::*;
+use sp_core::hashing::sha2_256;
+
+type Root = H256;
+type Domain = H256;
+type ValidatorIndex = u64;
+type ProofBranch  = Vec<Vec<u8>>;
+type ForkVersion = [u8; 4];
+
+const SLOTS_PER_EPOCH: u64 = 32;
+
+const EPOCHS_PER_SYNC_COMMITTEE_PERIOD: u64 = 256;
+
+const CURRENT_SYNC_COMMITTEE_INDEX: u64 = 22;
+const CURRENT_SYNC_COMMITTEE_DEPTH: u64 = 5;
+
+const NEXT_SYNC_COMMITTEE_DEPTH: u64 = 5;
+const NEXT_SYNC_COMMITTEE_INDEX: u64 = 23;
+
+const FINALIZED_ROOT_DEPTH: u64 = 6;
+const FINALIZED_ROOT_INDEX: u64 = 41;
+
+const MIN_SYNC_COMMITTEE_PARTICIPANTS: u64 = 1;
+
+/// GENESIS_FORK_VERSION('0x00000000')
+const GENESIS_FORK_VERSION: ForkVersion = [30, 30, 30, 30];
+
+/// DomainType('0x07000000')
+/// https://github.com/ethereum/consensus-specs/blob/dev/specs/altair/beacon-chain.md#domain-types
+const DOMAIN_SYNC_COMMITTEE: [u8; 4] = [7, 0, 0, 0];
+
+/// Beacon block header as it is stored in the runtime storage.
+#[derive(Clone, Default, Encode, Decode, PartialEq, RuntimeDebug, TypeInfo)]
+pub struct BeaconBlockHeader {
+	// The slot for which this block is created. Must be greater than the slot of the block defined by parentRoot.
+	pub slot: u64,
+	// The index of the validator that proposed the block.
+	pub proposer_index: ValidatorIndex,
+	// The block root of the parent block, forming a block chain.
+	pub parent_root: Root,
+	// The hash root of the post state of running the state transition through this block.
+	pub state_root: Root,
+	// The hash root of the Eth1 block
+	pub body_root: Root,
+}
+
+/// Sync committee as it is stored in the runtime storage.
+#[derive(
+	Clone, Default, Encode, Decode, PartialEq, RuntimeDebug, TypeInfo, Serialize, Deserialize
+)]
+pub struct SyncCommittee {
```

Comment:
Can optionally derive the Serialize and Deserialize traits with:

```suggestion
#[derive(
	Clone, Default, Encode, Decode, PartialEq, RuntimeDebug, TypeInfo,
)]
#[cfg_attr(feature = "std", derive(serde::Serialize, serde::Deserialize))]
```